src/main/java/com/eerussianguy/blazemap/feature/maps/AtlasExporter.java

```diff
@@ -0,0 +1,144 @@
+import net.minecraft.world.level.Level;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AtlasExporter {
+    private static Task task = null;
+
+    public static synchronized void export(Task task) {
+        if(AtlasExporter.task == null) {
+            AtlasExporter.task = task;
+            BlazeMapAsync.instance().clientChain.runOnDataThread(AtlasExporter::export);
+        }
+    }
+
+    public static synchronized Task getTask() {
+        return task;
+    }
+
+    private static void export() {
+        TileResolution resolution = TileResolution.FULL;
+        StorageAccess.Internal storage = BlazeMapClientEngine.getDimensionStorage(task.dimension);
+        AtlasInfo atlasInfo = new AtlasInfo();
+        NativeImage atlas = constructAtlas(storage, resolution, atlasInfo);
+
+        try(atlas) {
+            var layerKeys = task.map.value().getLayers();
+            for(var layerKey : layerKeys) {
+                if(!task.layers.contains(layerKey)) continue;
+                File folder = storage.getMipmap(layerKey.location, ".", resolution);
+                for(int regionX = atlasInfo.atlasStartX; regionX <= atlasInfo.atlasEndX; regionX++) {
+                    for(int regionZ = atlasInfo.atlasStartZ; regionZ <= atlasInfo.atlasEndZ; regionZ++) {
+                        File file = new File(folder, LayerRegionTile.getImageName(new RegionPos(regionX, regionZ)));
+                        if(!file.exists()) continue;
+                        NativeImage tile = NativeImage.read(Files.newInputStream(file.toPath()));
+
+                        int regionOffsetX = (regionX - atlasInfo.atlasStartX) * resolution.regionWidth;
+                        int regionOffsetZ = (regionZ - atlasInfo.atlasStartZ) * resolution.regionWidth;
+                        for(int x = 0; x < resolution.regionWidth; x++) {
+                            for(int z = 0; z < resolution.regionWidth; z++) {
+                                int atlasPixelX = regionOffsetX + x;
+                                int atlasPixelZ = regionOffsetZ + z;
+                                int atlasPixel = atlas.getPixelRGBA(atlasPixelX, atlasPixelZ);
+                                int tilePixel = tile.getPixelRGBA(x, z);
+                                atlas.setPixelRGBA(atlasPixelX, atlasPixelZ, Colors.layerBlend(atlasPixel, tilePixel));
+                            }
+                        }
+
+                        // non-atomic op on volatile int is ok because only 1 thread writes to variable
+                        // Java guarantees r/w access to 32-bit variables is atomic, so there are no mid-write reads
+                        task.tilesCurrent++;
+                    }
+                }
+            }
+
+            File file = new File(Minecraft.getInstance().gameDirectory, "screenshots/blazemap-export-"+ (System.currentTimeMillis() / 1000L) +".png");
```

Comment:
Nit: For purely file sorting purposes, we should probably try and keep the same file naming system as normal screenshots at the start of the file name and only put `blazemap-export` on the end of it. That means, when I'm grabbing screenshots to share in Discord or whatever, the export will always be grouped alphabetically with the rest of my most recent screenshots and so much easier to find!

Reply:
Fair point, will need to look up the format.

Reply:
Done in working branch